### ISSUE TITLE
support additional repositories in %AddDeps

### ISSUE THREAD

**Issue by ittayd** [ibm-et/spark-kernel#169]
Currently it supports only   https://repo1.maven.org/maven2/, but not all projects deploy there.

**Comment by lbustelo** [comment]
Yeah... I agree that we may want to allow users to pull from other repos.

**Comment by glyfnet** [comment]
I have implemented a AddResolver magic as well as modified the IvyDependencyDownloader to support a Chain of resolvers. If you add me as a collaborator, and I will push the branch i made the changes up and create a pull request.

**Comment by glyfnet** [comment]
```scala
%AddResolver ibiblio http://mirrors.ibiblio.org/maven2
```

    added Resolver ibiblio : http://mirrors.ibiblio.org/maven2



```scala
%AddDeps com.databricks spark-csv_2.10 1.2.0 --transitive
```

    :: loading settings :: url = jar:file:/opt/home/egreene/local/kernel/kernel-0.1.5-SNAPSHOT/lib/ivy-2.4.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
    :: resolving dependencies :: com.ibm.spark#spark-kernel;working
    	confs: [default]
    	found com.databricks#spark-csv_2.10;1.2.0 in ibiblio
    	found org.apache.commons#commons-csv;1.1 in ibiblio
    	found com.univocity#univocity-parsers;1.5.1 in ibiblio
    :: resolution report :: resolve 239ms :: artifacts dl 12ms
    	:: modules in use:
    	com.databricks#spark-csv_2.10;1.2.0 from ibiblio in [default]
    	com.univocity#univocity-parsers;1.5.1 from ibiblio in [default]
    	org.apache.commons#commons-csv;1.1 from ibiblio in [default]
    	---------------------------------------------------------------------
    	|                  |            modules            ||   artifacts   |
    	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
    	---------------------------------------------------------------------
    	|      default     |   3   |   0   |   0   |   0   ||   3   |   0   |
    	---------------------------------------------------------------------
    :: retrieving :: com.ibm.spark#spark-kernel
    	confs: [default]
    	0 artifacts copied, 3 already retrieved (0kB/10ms)

**Comment by EronWright** [comment]
In addition to making it more open-ended with `AddResolver`, please ensure that the default set of resolvers matches that of `spark-shell`.      Those include the local Maven and Ivy repositories and the official Spark Packages repository (bintray).

[core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L824](https://github.com/apache/spark/blob/branch-1.5/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L824)
